velox/type/Variant.h

```diff
@@ -108,6 +115,50 @@ struct VariantTypeTraits<TypeKind::ARRAY> {
   using stored_type = std::vector<variant>;
 };
 
+/// Variants contain a TypeKind and a value.
+/// DECIMAL type variants use TypeKind to store the kind, and this struct as the
+/// value to store the unscaled value, precision, scale, and null information.
+template <typename T>
+struct DecimalCapsule {
+  T value;
+  int precision;
+  int scale;
+  bool isNull = false;
```

Comment:
Any particular reason not to combine value and isNull into an std::optional<int>?

Reply:
We can serialize a variant via variant::serialize(), (not implemented in this PR). It seemed to be that `std::optional` is not supported.

Reply:
This is custom logic, no? What would prevent us from supporting an optional there?

Reply:
You are right. I will fix this.